### PR TITLE
Add Istio 1.2 conformance report

### PR DESCRIPTION
**Pull request by howardjohn**
**What type of PR is this?**
/area conformance

**What this PR does / why we need it**:
Shows our support for v1.2

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: conformance/reports/v1.2.0/istio-istio/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Istio
+
+## Table of Contents
+
+|API channel|Implementation version|Mode|Report|
+|-----------|----------------------|----|------|
+|x|[1.24](https://github.com/istio/istio/releases/tag/1.24.0)|x|[link](./experimental-1.24-default-report.yaml)|
```
[the remark this answers is not in the document]
Comment (arkodg) [reply conformance/reports/v1.2.0/istio-istio/README.md:7]: my suggestion would be to link this to an existing tag like https://github.com/istio/istio/releases/tag/1.24.0-alpha.0 and edit it once the release is done

[the remark this answers is not in the document]
Comment (mikemorris) [reply conformance/reports/v1.2.0/istio-istio/README.md:7]:
```suggestion
|x|[1.24](https://github.com/istio/istio/releases/tag/1.24.0-alpha.0)|x|[link](./experimental-1.24-default-report.yaml)|
```

File: conformance/reports/v1.2.0/istio-istio/experimental-1.24-default-report.yaml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+date: "2024-10-20T03:00:03Z"
+gatewayAPIChannel: experimental
+gatewayAPIVersion: v1.2.0
+implementation:
+  contact:
+  - '@istio/maintainers'
+  organization: istio
+  project: istio
+  url: https://istio.io/
+  version: "1.24"
```
Comment (arkodg) [inline conformance/reports/v1.2.0/istio-istio/experimental-1.24-default-report.yaml:11]:
```suggestion
  version: 1.24.0-alpha.0
```

**Comment by arkodg** [comment]
seeing an error
```
./conformance/reports/v1.2.0/istio-istio/experimental-1.24-default-report.yaml
  125:24    error    no new line character at the end of file  (new-line-at-end-of-file)
 [0;31mTest FAILED: hack/../hack/verify-yamllint.sh 
 [0;31mSummary of failed tests:
[0;31m- hack/../hack/verify-yamllint.sh
make: *** [Makefile:117: verify] Error 1
```
cc @mikemorris @howardjohn

File: conformance/reports/v1.2.0/istio-istio/experimental-1.24-default-report.yaml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    - HTTPRouteRequestTimeout
+    - HTTPRouteResponseHeaderModification
+    - HTTPRouteSchemeRedirect
+    - MeshConsumerRoute
+    unsupportedFeatures:
+    - MeshClusterIPMatching
+  name: MESH-HTTP
+  summary: Core tests succeeded. Extended tests succeeded.
+succeededProvisionalTests:
+- GatewayInfrastructure
```
Comment (mikemorris) [inline conformance/reports/v1.2.0/istio-istio/experimental-1.24-default-report.yaml:125]:
```suggestion
- GatewayInfrastructure

```

**Pull request merged**